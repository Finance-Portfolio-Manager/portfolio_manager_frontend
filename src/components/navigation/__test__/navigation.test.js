import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import Navigation from "../navigation";

import { shallow, ShallowWrapper, mount } from "enzyme";

import renderer from "react-test-renderer";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import NavDropDown from "../navdropdown";

import React, { useState } from 'react';

configure({ adapter: new Adapter() });
const wrapper = shallow(<Navigation></Navigation>);

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

test("matches snapshot", ()=>{
  const tree = renderer.create(<Navigation></Navigation>).toJSON();
  expect(tree).toMatchSnapshot();
})

test("has Portfolios item", () => {
  act(() => {
    render(<Navigation />, container);
  });
  expect(container.querySelector('div > span').firstChild.nodeValue).toEqual("Portfolios");
});


describe("Nav bar rendering", ()=>{
  let component = shallow(<Navigation></Navigation>);

  test("renders a title", ()=>{
    expect(component.find("a").first().text()).toContain('Team Name');
  })

  test("renders two dropdown menus", ()=>{
    expect(component.find("NavDropDown").length).toBe(2);
  })

  test("renders Home", ()=>{
    expect(component.find("a").at(1).text()).toContain('Home');
  })

  test("renders Your Accounts", ()=>{
    expect(component.find("a").at(2).text()).toContain('Your Accounts');
  })

  test("renders Favorite", ()=>{
    expect(component.find("a").at(3).text()).toContain('Favorite');
  })

  test("renders Public Portfolio", ()=>{
    expect(component.find("a").at(4).text()).toContain('Public portfolio');
  })

  test("renders About Us", ()=>{
    expect(component.find("a").at(5).text()).toContain('About Us');
  })

  test("renders Pie Chart", ()=>{
    expect(component.find("a").at(6).text()).toContain('Pie Chart');
  })

  test("renders Stock Prices", ()=>{
    expect(component.find("a").at(7).text()).toContain('Stock Prices');
  })

  test("renders Stock Graph", ()=>{
    expect(component.find("a").at(8).text()).toContain('Stock Graph');
  })

  test("renders Account", ()=>{
    expect(component.find("a").at(9).text()).toContain('Account');
  })

  test("renders Log in", ()=>{
    expect(component.find("a").at(10).text()).toContain('Log in');
  })

  test("renders Sign up", ()=>{
    expect(component.find("a").at(11).text()).toContain('Sign up');
  })

  test("renders Log Out", ()=>{
    expect(component.find("a").last().text()).toContain('Log Out');
  })
})

describe("navigation response to actions", ()=>{
  test("dropdown menus don't crash when clicked", ()=>{
    const component = shallow(<Navigation></Navigation>).find("NavDropDown");
    expect(component.length).toBe(2);

    let clickEvent1 = () =>{
      component.first().simulate('click');
    }
    let clickEvent2 = () =>{
      component.at(1).simulate('click');
    }

    expect(clickEvent1).not.toThrow();
    expect(clickEvent2).not.toThrow();
  })
})
