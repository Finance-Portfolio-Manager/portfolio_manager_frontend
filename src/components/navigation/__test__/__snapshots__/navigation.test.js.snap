// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container-fluid p-0"
>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark primary-color"
  >
    <img
      alt="Logo"
      className="m-2"
      src="apexstocks.jpg"
    />
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <div
        className=" d-flex flex-column flex-md-row justify-content-between flex-fill"
      >
        <ul
          className="navbar-nav primary-text"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <div
              className="text-white py-0 border-0 dropdown"
              onMouseLeave={[Function]}
            >
              <div
                className="btn rounded-0 "
                href=""
                onMouseEnter={[Function]}
              >
                <span
                  className="text-white"
                >
                  Portfolios
                </span>
              </div>
            </div>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/about"
            >
              About Us
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active align-item-center"
              href="/new-transaction"
            >
              Transaction
            </a>
          </li>
        </ul>
        <div
          className="d-flex align-self-center me-3"
        >
          <span
            className="align-self-center p1"
          >
            <i
              className="bi-person-circle text-white"
            />
          </span>
          <div
            className="navbar-nav dropdown"
          >
            <a
              aria-expanded="false"
              className="nav-link dropdown-toggle active"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdown"
              role="button"
            >
              Account
            </a>
            <ul
              aria-labelledby="navbarDropdown"
              className="dropdown-menu dropdown-menu-end dropdown-menu-gray"
            >
              <li>
                <a
                  className="dropdown-item"
                  href="/login"
                >
                  Log in
                </a>
              </li>
              <li>
                <a
                  className="dropdown-item"
                  href="/register"
                >
                  Sign up
                </a>
              </li>
              <li>
                <a
                  className="dropdown-item"
                  href="/home"
                >
                  Log Out
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
